### PR TITLE
fix: #1391 Fix export issue on module with subentries

### PR DESCRIPTION
**Pull request by cpaulve-1A**

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -66,7 +66,6 @@
       "@o3r/stylelint-plugin": ["packages/@o3r/stylelint-plugin/src"],
       "@o3r/styling": ["packages/@o3r/styling/src/public_api"],
       "@o3r/testing": ["packages/@o3r/testing/src/public_api"],
-      "@o3r/testing/*": ["packages/@o3r/testing/src/*"],
```
Comment (kpanot) [inline tsconfig.base.json:69]: Not sure you should remove it.
Without this rules, the deep imports will be rerouted to src/ folder in this monorepo

File: packages/@ama-sdk/swagger-builder/package.json [hunk 2]
```diff
@@ -10,14 +10,14 @@
     "nx": "nx",
     "ng": "yarn nx",
     "build": "yarn nx build ama-sdk-swagger-builder",
-    "postbuild": "patch-package-json-main",
+    "postbuild": "patch-package-json-main && cpy 'src/schemas/*.json' 'dist/schemas'",
```
Comment (kpanot) [inline packages/@ama-sdk/swagger-builder/package.json:13]: is `src/schemas/*.json` already included in Nx task outputs?

File: packages/@ama-sdk/core/package.json [hunk 3]
```diff
@@ -2,27 +2,13 @@
...
+  "module": "dist/public_api.js",
   "esm2015": "dist/esm2015/public_api.js",
-  "esm2020": "dist/src/public_api.js",
-  "typings": "dist/src/public_api.d.ts",
+  "esm2020": "dist/public_api.js",
+  "typings": "dist/public_api.d.ts",
+  "main": "dist/public_api.js",
+  "types": "dist/public_api.d.ts",
   "sideEffects": false,
-  "exports": {
```
Comment (kpanot) [inline packages/@ama-sdk/core/package.json:10]: Not sure you should remove these exports.
The purpose of it is to be able to use this package (and then the SDK) in a esm context on node.
Why did you remove it here? Is it causing issue somewhere?

File: packages/@ama-sdk/swagger-builder/package.json (hunk 2 above)
Comment (cpaulve-1A) [reply packages/@ama-sdk/swagger-builder/package.json:13]: I want to remove the src folder from the dist. Hence, the one from nx will be removed.

File: packages/@ama-sdk/generator-sdk/tsconfig.build.json [hunk 4]
```diff
@@ -7,6 +7,7 @@
     "moduleResolution": "node",
     "noEmitHelpers": false,
     "outDir": "dist/",
+     "rootDir": "src",
```
Comment (impy88) [inline packages/@ama-sdk/generator-sdk/tsconfig.build.json:10]: space :)

File: nx.json [hunk 5]
```diff
@@ -152,8 +152,8 @@
         "jest-fixtures"
       ],
       "outputs": [
-        "{projectRoot}/dist/**/fixtures/jasmine",
-        "{projectRoot}/dist/**/fixtures/jest"
+        "{projectRoot}/dist/fixtures/jasmine",
```
Comment (impy88) [inline nx.json:155]: Do you keep build fixtures, so jest and jasmine seem mixed because of that...

Comment (cpaulve-1A) [reply nx.json:157]: I misunderstood your remark, we still build both but they are exported via different entrypoints. 
We should not have reference of jest in jasmine and vice-versa

File: packages/@o3r/analytics/project.json [hunk 6]
```diff
@@ -22,11 +22,27 @@
...
       "options": {
-        "project": "packages/@o3r/analytics/ng-package.json",
+        "project": "packages/@o3r/analytics/src/ng-package.json",
         "tsConfig": "packages/@o3r/analytics/tsconfig.build.json"
       }
     },
+    "prepare": {
+      "executor": "nx:run-script",
+      "outputs": [
+        "{projectRoot}/src/package.json"
```
Comment (kpanot) [inline packages/@o3r/analytics/project.json:37]: maybe you can also specify input as it is only 1 file

File: packages/@o3r/testing/package.json [hunk 7]
```diff
@@ -6,85 +6,87 @@
     "testing",
     "otter"
   ],
+  "main": "dist/public_api.js",
+  "typings": "dist/public_api.d.ts",
+  "ng-update": {
+    "migrations": "./src/migration.json"
```
Comment (kpanot) [inline packages/@o3r/testing/package.json:12]: should not be in `./migration.json`?

Comment (cpaulve-1A) [reply packages/@o3r/testing/package.json]: updated

File: packages/@ama-sdk/generator-sdk/package.json [hunk 8]
```diff
@@ -78,5 +86,7 @@
     "node": ">=14.0.0",
     "npm": ">=4",
     "yarn": "<4.0.0"
-  }
+  },
+  "main": "./dist/generators/sdk-generator.js",
```
Comment (mrednic-1A) [inline packages/@ama-sdk/generator-sdk/package.json:90]: I think it's without `./dist` and I am not sure it is needed

File: packages/@o3r/testing/package.json [hunk 9]
```diff
@@ -6,85 +6,87 @@
     "testing",
     "otter"
   ],
+  "main": "dist/public_api.js",
```
Comment (mrednic-1A) [inline packages/@o3r/testing/package.json:9]: `dist` should not be needed as in the published package we dont have a dist folder.

Comment (cpaulve-1A) [reply packages/@o3r/testing/package.json:9]: It is updated during the postbuild task which goes over all the dist reference and remove them

File: packages/@ama-sdk/generator-sdk/package.json (hunk 8 above)
Comment (cpaulve-1A) [reply packages/@ama-sdk/generator-sdk/package.json:90]: Yeah it can probably be removed as this is not the way the generator shall be imported

File: packages/@o3r/testing/package.json (hunk 9 above)
Comment (cpaulve-1A) [reply packages/@o3r/testing/package.json]: Well it does not change anything after tests, I will leave it this way :)

**Review by kpanot — DISMISSED** [review]

**Review by kpanot — APPROVED** [review]

**Review by matthieu-crouzet — APPROVED** [review]

**Review by mrednic-1A — APPROVED** [review]

**Review by kpanot — APPROVED** [review]
